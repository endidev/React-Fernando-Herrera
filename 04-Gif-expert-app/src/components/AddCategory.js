import React, {useState} from 'react'
import PropTypes from 'prop-types';

export const AddCategory = ({setCategories}) => {

    const [inputValue, setInputValue] = useState('')

    const handleInputChange = (e) => {
        setInputValue( e.target.value )
    }

    const handdleSubmit = (e) => {
        e.preventDefault();
        
        if(inputValue.trim().length > 2){
            setCategories( cats => [inputValue, ...cats])
        }

    }

    return (
        <form onSubmit={handdleSubmit}>
        
            <input 
                type='text'
                value= {inputValue}
                onChange={ handleInputChange }
            />
        </form>
    )
}


AddCategory.propTypes = {
    setCategories: PropTypes.func.isRequired
}
