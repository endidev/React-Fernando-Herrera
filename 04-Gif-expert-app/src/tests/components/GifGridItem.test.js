import PropTypes from 'prop-types'


import { shallow } from "enzyme"
import { GifGridItem } from "../../components/GifGridItem"


describe('Componente Gif Grid Item', () =>{

    
    const title = 'Un titulo';
    const url = 'https://localhost/algo.jpg';
    
    let wrapper = shallow(<GifGridItem title={title} url={url}/>)
    

    test('Mostrar el componente correctamente', () => {
        expect( wrapper ).toMatchSnapshot();
    })
    
    test('Debe tener un texto con el titulo', () => {
        const p = wrapper.find('p');
        expect( p.text().trim() ).toBe( title );
    })

    test('debe tener la imagen, con url y alt, de las propiedades enviadas', () => {
        const img = wrapper.find('img');
        // console.log(img.props())
        expect( img.prop('src') ).toBe(url);
        expect( img.prop('alt') ).toBe(title);
    })
    
    test('debe tener el animate__fadeIn', () => {
        const div = wrapper.find('div')
        const testear = div.prop('className').includes('animate__fadeIn');

        expect( testear ).toBe(true);
    })
    
    
})