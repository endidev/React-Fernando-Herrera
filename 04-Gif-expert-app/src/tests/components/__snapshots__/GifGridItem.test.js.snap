// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Componente Gif Grid Item Mostrar el componente correctamente 1`] = `
<div
  className="card animate__animated animate__fadeIn"
>
  <img
    alt="Un titulo"
    src="https://localhost/algo.jpg"
  />
  <p>
    Un titulo
  </p>
</div>
`;
